.github/workflows/dojo-e2e.yml

```diff
@@ -0,0 +1,90 @@
+name: e2e
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  e2e:
+    name: E2E Tests
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout code
+      uses: actions/checkout@v4
+
+    - name: Set up Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: '22'
+
+    - name: Install pnpm
+      uses: pnpm/action-setup@v4
+      with:
+        version: 9.0.0
+
+    - name: Install Poetry
+      uses: snok/install-poetry@v1
+      with:
+        version: latest
+        virtualenvs-create: true
+        virtualenvs-in-project: true
+
+    - name: Install uv
+      uses: astral-sh/setup-uv@v6
+
+    - name: Setup pnpm cache
+      uses: actions/cache@v4
+      with:
+        path: ~/.local/share/pnpm/store
+        key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+        restore-keys: |
+          ${{ runner.os }}-pnpm-store-
+
+    - name: Install dependencies
+      working-directory: typescript-sdk
+      run: pnpm install --frozen-lockfile
+
+    - name: Prepare dojo for e2e
+      working-directory: typescript-sdk/apps/dojo
+      run: node ./scripts/prep-dojo-everything.js -e2e
+
+    - name: Install e2e dependencies
+      working-directory: typescript-sdk/apps/dojo/e2e2
+      run: |
+        pnpm install --frozen-lockfile
+        pnpm dlx playwright install --with-deps
+
+    - name: write langgraph env files
+      working-directory: typescript-sdk/integrations/langgraph
+      env:
+        OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+        LANGSMITH_API_KEY: ${{ secrets.LANGSMITH_API_KEY }}
+      run: |
+        echo "OPENAI_API_KEY=${OPENAI_API_KEY}" > examples/python/.env
+        echo "LANGSMITH_API_KEY=${LANGSMITH_API_KEY}" >> examples/python/.env
+        echo "OPENAI_API_KEY=${OPENAI_API_KEY}" > examples/typescript/.env
+        echo "LANGSMITH_API_KEY=${LANGSMITH_API_KEY}" >> examples/typescript/.env
+        echo "OPENAI_API_KEY=${OPENAI_API_KEY}" > python/ag_ui_langgraph/.env
+        echo "LANGSMITH_API_KEY=${LANGSMITH_API_KEY}" >> python/ag_ui_langgraph/.env
+
+    - name: Run dojo+agents and tests
+      working-directory: typescript-sdk/apps/dojo/e2e2
+      env:
+        OPENAI_API_KEY: ${{ secrets.OPENAI_API_KEY }}
+        LANGSMITH_API_KEY: ${{ secrets.LANGSMITH_API_KEY }}
+      run: |
+        node ../scripts/run-dojo-everything.js &
+        npx wait-port 9999
+        sleep 10
```

Comment:
Does this leave us open to flakey tests?

Reply:
It _could_ but so far even in CI everything is running in <5 seconds from dojo coming up, so I figured 10 seconds is a big enough buffer. 

Ideally we can add a health check to all of the agents and make sure they're all running before we move on, but I didn't want to block on that.